misk/src/main/kotlin/misk/tracing/interceptors/TracingMethodInterceptor.kt

```diff
@@ -0,0 +1,34 @@
+package misk.tracing.interceptors
+
+import io.opentracing.Scope
+import io.opentracing.Tracer
+import io.opentracing.tag.Tags
+import misk.logging.getLogger
+import org.aopalliance.intercept.MethodInterceptor
+import org.aopalliance.intercept.MethodInvocation
+import javax.inject.Inject
+import javax.inject.Provider
+
+private val logger = getLogger<TracingMethodInterceptor>()
+
+class TracingMethodInterceptor @Inject internal constructor(
+  private val tracerProvider: Provider<Tracer>
+) : MethodInterceptor {
+  override fun invoke(invocation: MethodInvocation): Any? {
+    var result: Any? = null
+    var scope: Scope? = null
+    val tracer = tracerProvider.get()
+
+    try {
+      scope = tracer.buildSpan("${invocation.javaClass.simpleName}.${invocation.method.name}").startActive(true)
+      result = invocation.proceed()
+    } catch (exception: Exception) {
+      logger.warn("failed attempting to trace method", exception)
```

Comment:
do we really want this logged every time a traced method throws an exception? seems like a lot of noise and the log message reads like there is a problem with the tracing, not the thing being traced

Reply:
Yeah, that's a good point. I think this is wrong in general. At a minimum we should be re-throwing the exception so that the tracer doesn't swallow it. Think I'm going to kill this catch altogether.